8BitCalculator/8BitCalculator/main.asm

```diff
@@ -0,0 +1,364 @@
+            
+        SET_TO_ZERO_BUTTON:                     ; The handler that detects if the SET_TO_ZERO button has been pressed
+            IN      TEMP, PINC                  ; Read the value from Port C and put it into TEMP
+            CPI     TEMP, STZB_ON               ; Compare TEMP and SET_TO_ZERO_BUTTON_ON
+            BRNE    STZ_CONTINUE                ; If the above values are not equal that means the button is not pressed and we can skip the call to its operation
+            CALL    SET_TO_ZERO                 ; Call the SET_TO_ZERO subroutine
+        STZ_CONTINUE:                           ; Used to skip the call
+
+        OP_PLUS_BUTTON:                         ; The handler that detects if the OPERATION_PLUS_BUTTON has been pressed
+            IN      TEMP, PINC                  ; Read the value from Port C and put it into TEMP
+            CPI     TEMP, OPPB_ON               ; Compare TEMP and OPERATION_PLUS_BUTTON_ON
+            BRNE    OPP_CONTINUE                ; If the above values are not equal that means the button is not pressed and we can skip the call to its operation
+            CALL    OP_PLUS                     ; Call the OPERATION_PLUS subroutine
+        OPP_CONTINUE:                           ; Used to skip the call
+        
+        
+        OP_SUBSTRACT_BUTTON:                    ; The handler that detects if the OPERATION_SUBSTRACT_BUTTON has been pressed
+            IN      TEMP, PINC                  ; Read the value from Port C and put it into TEMP
+            CPI     TEMP, OPSB_ON               ; Compare TEMP and OPERATION_PLUS_BUTTON_ON
+            BRNE    OPS_CONTINUE                ; If the above values are not equal that means the button is not pressed and we can skip the call to its operation
+            CALL    OP_SUB                      ; Call the OPERATION_SUBSTRACT subroutine
+        OPS_CONTINUE:
+
+    CONTINUE_MAIN_LOOP_DELAY:                   ; Used to continue the main loop delay after one of the setter buttons has been pressed
+        DEC     ML_INNER                        ; Decrement the inner loop value
+        BRNE    MAIN_LOOP_DELAY                 ; ^
+        DEC     ML_MIDER                        ; ^
+        BRNE    MAIN_LOOP_DELAY                 ; ^
+        DEC     ML_OUTER                        ; ^
+        BRNE    MAIN_LOOP_DELAY                 ; ^
+        NOP                                     ; Extra cycle needed to make it 250ms, does nothing
+    
+    RJMP    MAIN_LOOP                           ; Go back to the start of the MAIN_LOOP after the delay is finished
+
+
+SET_TO_ONE:                                     ; The subroutine that deals with the instructions the SET_TO_ONE button has to execute
+    CALL    DEBOUNCE                            ; Call DEBOUNCE (just another 20ms delay) in order to stabilize the input we get from the button, otherwise you get inconsistent behaviour

```

Comment:
You might be able to simplify this subroutine considerably by using the logic shift operations (`lsl`and `lsr`) instead of having code dealing with each bit position separately.